src/tools/kata-ctl/src/check.rs

```diff
@@ -129,6 +133,55 @@ pub fn run_network_checks() -> Result<()> {
     Ok(())
 }
 
+// Set of basic checks for kvm. Architectures should implement more specific checks if needed
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kvm_is_usable_generic(kvm_device: &str, machine_type: i32) -> Result<()> {
+    // check for root user
+    if !nix::unistd::Uid::effective().is_root() {
+        return Err(anyhow!("Will not perform kvm checks as non root user"));
+    }
+
+    // we do not want to erroneously create syscalls to any device besides /dev/kvm
+    if kvm_device != "/dev/kvm" {
```

Comment:
@amshinde I could remove it, but then we really have no unit tests for it

Reply:
Also on rereading the documentation, I'm going to remove the vm type and enforce default, if an arch needs something more in depth it can be implemented separately

Reply:
updated